### ISSUE TITLE
php-imagick upgrade fails for version 3.4.3-4

### ISSUE THREAD

**Issue by langemeijer** [oerdnj/deb.sury.org#1003]
When I upgrade the php-imagick package I am greeted with this error:

> php_invoke: module imagick not supported in preinst for PHP 7.1 apache2 sapi

I'm using your 7.1.24 php packages. The 3.4.3-3 seems to work fine (it shows with `php -m | grep imagick`)

```
root@hostname:~# aptitude full-upgrade
The following packages will be upgraded: 
  php-imagick 
1 packages upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/236 kB of archives. After unpacking 456 kB will be used.
Do you want to continue? [Y/n/?] 
(Reading database ... 46700 files and directories currently installed.)
Preparing to unpack .../php-imagick_3.4.3-4+0~20181107094741.8+jessie~1.gbpba1eeb_amd64.deb ...
php_invoke: module imagick not supported in preinst for PHP 7.1 apache2 sapi
dpkg: error processing archive /var/cache/apt/archives/php-imagick_3.4.3-4+0~20181107094741.8+jessie~1.gbpba1eeb_amd64.deb (--unpack):
 subprocess new pre-installation script returned error exit status 1
Errors were encountered while processing:
 /var/cache/apt/archives/php-imagick_3.4.3-4+0~20181107094741.8+jessie~1.gbpba1eeb_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
Failed to perform requested operation on package.  Trying to recover:

root@hostname:~#
```

**Comment by threesquared** [comment]
I am also experiencing this error but with the `php-redis` package:

```
root@eynhallow:~# apt-get install php-redis
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following packages will be upgraded:
  php-redis
1 upgraded, 0 newly installed, 0 to remove and 165 not upgraded.
Need to get 0 B/358 kB of archives.
After this operation, 207 kB of additional disk space will be used.
(Reading database ... 139829 files and directories currently installed.)
Preparing to unpack .../php-redis_4.2.0-1+ubuntu16.04.1+deb.sury.org+1_amd64.deb ...
php_invoke: module redis not supported in preinst for PHP 7.1 fpm SAPI
dpkg: error processing archive /var/cache/apt/archives/php-redis_4.2.0-1+ubuntu16.04.1+deb.sury.org+1_amd64.deb (--unpack):
 subprocess new pre-installation script returned error exit status 1
Errors were encountered while processing:
 /var/cache/apt/archives/php-redis_4.2.0-1+ubuntu16.04.1+deb.sury.org+1_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

**Comment by oerdnj** [comment]
Could you try first upgrading `php-common` package and then upgrade `php-redis`?

**Comment by threesquared** [comment]
That fixed it for me, just running `apt-get install php-common` before. Thanks!

**Comment by oerdnj** [comment]
Thanks for the confirmation, the extension packages will need Pre-Depends on the correct `php-common` version.

**Comment by oerdnj** [comment]
Ok, so updating `php-common` had nothing to do with that. It just doesn't fail on second try.  I already have a fix, but it requires a rebuild to all the packages, so it will take some time.

**Comment by svparijs** [comment]
Got the same error...

**Comment by juraisa** [comment]
> Ok, so updating `php-common` had nothing to do with that. It just doesn't fail on second try. I already have a fix, but it requires a rebuild to all the packages, so it will take some time.

Hi just wondering if there has been any update on the fix for this. The php-redis/php-imagick/php-mongodb/php-memcached packages all seem to have the same issue if attempting to do an upgrade on an older PHP version installation to the latest available packages, removing the packages in question and reinstalling them clean seems to work fine at least in the interim, but a direct upgrade path will be great longterm

**Comment by oerdnj** [comment]
The `php-common` and `dh-php` packages has been fixed and the individual packages should be fixed with each new version (which I expect will spike with PHP 7.3.0 now being gold).

**Comment by oerdnj** [comment]
Could you actually enlist the versions of packages that failed to upgrade?

**Comment by juraisa** [comment]
> Could you actually enlist the versions of packages that failed to upgrade?

sure, I upgraded a PHP 7.2 test system to the latest available 7.2 packages to make sure things were reflective of the current repo packages

`ii  libapache2-mod-php7.2                   7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        server-side, HTML-embedded scripting language (Apache 2 module)
ii  php7.2-bcmath                           7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        Bcmath module for PHP
ii  php7.2-cli                              7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        command-line interpreter for the PHP scripting language
ii  php7.2-common                           7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        documentation, examples and common module for PHP
ii  php7.2-curl                             7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        CURL module for PHP
ii  php7.2-gd                               7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        GD module for PHP
ii  php7.2-gmp                              7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        GMP module for PHP
ii  php7.2-imap                             7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        IMAP module for PHP
ii  php7.2-intl                             7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        Internationalisation module for PHP
ii  php7.2-json                             7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        JSON module for PHP
ii  php7.2-ldap                             7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        LDAP module for PHP
ii  php7.2-mbstring                         7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        MBSTRING module for PHP
ii  php7.2-mysql                            7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        MySQL module for PHP
ii  php7.2-odbc                             7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        ODBC module for PHP
ii  php7.2-opcache                          7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        Zend OpCache module for PHP
ii  php7.2-pgsql                            7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        PostgreSQL module for PHP
ii  php7.2-pspell                           7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        pspell module for PHP
ii  php7.2-readline                         7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        readline module for PHP
ii  php7.2-recode                           7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        recode module for PHP
ii  php7.2-soap                             7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        SOAP module for PHP
ii  php7.2-sqlite3                          7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        SQLite3 module for PHP
ii  php7.2-sybase                           7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        Sybase module for PHP
ii  php7.2-tidy                             7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        tidy module for PHP
ii  php7.2-xml                              7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        DOM, SimpleXML, WDDX, XML, and XSL module for PHP
ii  php7.2-zip                              7.2.13-1+0~20181207100623.13+jessie~1.gbpf57305      amd64        Zip module for PHP`

I then installed the version of php-mongodb currently in use in my production environment:
`~# dpkg -i php-mongodb_1.4.2-1+0~20180321135644.5+jessie~1.gbp34313a_amd64.deb 
Selecting previously unselected package php-mongodb.
(Reading database ... 68026 files and directories currently installed.)
Preparing to unpack php-mongodb_1.4.2-1+0~20180321135644.5+jessie~1.gbp34313a_amd64.deb ...
Unpacking php-mongodb (1.4.2-1+0~20180321135644.5+jessie~1.gbp34313a) ...
Setting up php-mongodb (1.4.2-1+0~20180321135644.5+jessie~1.gbp34313a) ...`

I then attempted to install latest version of php-mongodb available in the repo in order to upgrade php-mongodb, and got the preinst error mentioned by OP

`~# dpkg -i php-mongodb_1.5.3-2+0~20181207132514.5+jessie~1.gbp43d844_amd64.deb 
(Reading database ... 68033 files and directories currently installed.)
Preparing to unpack php-mongodb_1.5.3-2+0~20181207132514.5+jessie~1.gbp43d844_amd64.deb ...
php_invoke: module mongodb not supported in preinst for PHP 7.2 cli sapi
dpkg: error processing archive php-mongodb_1.5.3-2+0~20181207132514.5+jessie~1.gbp43d844_amd64.deb (--install):
 subprocess new pre-installation script returned error exit status 1
Errors were encountered while processing:
 php-mongodb_1.5.3-2+0~20181207132514.5+jessie~1.gbp43d844_amd64.deb
`

If I do an apt-get remove of the php-mongodb 1.4.2-1 package, and then do the dpkg -i on the 1.5.3-2 package, it installs properly, but it does not want to upgrade in place with the 1.4.2-1 package installed

**Comment by oerdnj** [comment]
This bug is now a year old, just closing.
